internal/security-mail-automation/security-email-automation.gs

```diff
@@ -0,0 +1,278 @@
+*
+*@param userId Email address of the user
+*@param messageId Message ID of the email message
+*
+*@return returns an object containing the subject, from and cc of a selected email
+*/
+function checkMailInfo(userId, messageId) {  
+  var headers = Gmail.Users.Messages.get(userId, messageId).payload.headers;
+  var isWso2 = new Array(); 
+  
+  for each (var obj in headers) {
+    if (obj["name"] == "Subject")
+      var subject = obj["value"];
+    if (obj["name"] == "To")
+      var to = obj["value"];
+    if (obj["name"] == "From")
+      var from = obj["value"];
+    if (obj["name"] == "Cc")
+      var cc = obj["value"];
+  }
+     
+  var mailInfo = Object.create(null, {
+      subject:{
+        value: subject
+      },
+      from:{
+        value: from
+      },
+      cc:{
+        value:isWso2
+      }
+    });
+    return mailInfo;
+  } else {
+    cc = defaultCC; //if the CC section is null, sets a default cc to security@
+    
+    var mailInfo = Object.create(null, {
+      subject:{
+        value: subject
+      },
+      from:{
+        value: from
+      },
+      cc:{
+        value:cc
+      }
+    });
+    return mailInfo;
+  }  
+}
+
+/**
+* checkDomain() -- To check the domain of the recieved emails
+*
+*@param to Reciever of the email
+*@param from Sender of the email
+*@param cc  Array containing any CC participants of the email
+*
+*@return returns an array of wso2 employees by removing the users that are out of the wso2 domain
+*
+*/
+function checkDomain(to, from, cc) { 
+  var tempArray = new Array();
+  var isWso2 = new Array();
+  var isCustomer = new Array();
+  
+  var str = to + "," + from + "," + cc;
+  tempArray = str.split(",");
+  
+  for each (var item in tempArray) {
+   var check = item.split('<').pop().split('>')[0];
+   var regExp = new RegExp("^[a-zA-Z0-9_.+-]+@(?:(?:[a-zA-Z0-9-]+\.)?[a-zA-Z]+\.)?(wso2)\.com$");
+    
+    if (regExp.test(check) == true) { //checks whther the domain is equal to wso2.com 
+      isWso2.push(item);
+    } else {
+      isCustomer.push(item);
+    }
+  }
+  return isWso2;
+}
+
+/**
+* createIssue() -- Creates a JIRA issue ticket for each email recieved by stating the reporter, timestamp and 
+*			the subject of the email
+*
+*@param subject Subject of the email to set as the summary of the ticket
+*@param from Reciever of the email to fill the description
+*@param threadID Thread ID of the email
+*@param cc  Array containing any CC participants of the email
+*
+*@return returns the URL of the issue ticket created
+*/
+function createIssue(subject, from, threadID, cc) {
+  if (subject == null) { //if the subject is null 
+    subject = "Security Vulnerability";
+  }
+  var reporter = jiraUsername; 
+  var username = jiraUsername;
+  var token = jiraPassword;
+  var encCred = Utilities.base64Encode(username + ":" + token);
+  var url = "https://WSO2_JIRA_DOMAIN/jira/rest/api/2/issue/";
```

Comment:
Shall we read this url from config file? Because in future if we change the issue tracking platform, we can easily mention that in config file.